### PR TITLE
Recorder timing

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
The designs for some of the active steps have changed to remain on the step until the participant
taps a “Next” button. Because of this, the recorders can continue running for a long time. Stop
the recorders when the step timer indicates that the step is done recording.

To allow for testing this, the default active step nib and UI/UX was changed to the active step used
by the MotorControl module. Currently, the default nib is not used in any modules that we support.

https://sagebionetworks.jira.com/browse/MP2-191
Note: Additional fixes will be required in the MotionControl framework.

### CODE REVIEW

File: Research/ResearchUI/iOS/RSDStepViewController.swift [hunk 1]
```diff
@@ -1052,7 +1057,7 @@ open class RSDStepViewController : UIViewController, RSDStepController, RSDCance
             
             // Update the countdown
             if let stepDuration = self.activeStep?.duration {
-                countdown = Int(stepDuration - duration)
+                countdown = Int(stepDuration) - Int(floor(duration))
```
Comment (Erin-Mounts) [inline Research/ResearchUI/iOS/RSDStepViewController.swift]: unless duration can be negative, I think the floor() is redundant? maybe worth keeping for explicitness, but then stepDuration should be handled the same way

File: Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift [hunk 2]
```diff
@@ -77,11 +67,25 @@ open class RSDActiveStepViewController: RSDFullscreenImageStepViewController {
...
     /// A label that is updated to show a countdown. For example, "5:35".
     @IBOutlet open var countdownLabel: UILabel?
+    
+    /// A label that is displayed to the user when the countdown is finished.
+    @IBOutlet open var doneLabel: UILabel?
+    
+    /// Returns the allowed countdown units for the countdown label. Default is to show only seconds
+    /// if the duration is less than 90 seconds.
+    open var allowedCountdownUnits: NSCalendar.Unit {
+        if self.activeStep?.duration ?? 0 > 90 {
```
Comment (Erin-Mounts) [inline Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift]: I can never remember operator precedence for obscure-ish ones like ?? and >, so I'd like to see parentheses to make it explicit -- also makes it easier/faster for me to parse it even when I can eventually figure it out from context

File: Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift [hunk 3]
```diff
@@ -91,7 +95,40 @@ open class RSDActiveStepViewController: RSDFullscreenImageStepViewController {
...
+            unitLabel?.text = nil
+            countdownLabel?.text = nil
+            doneLabel?.isHidden = false
+        }
+        else {
+            doneLabel?.isHidden = true
             countdownLabel?.text = countdownFormatter.string(from: TimeInterval(countdown))
+            if self.allowedCountdownUnits == [.second] {
+                switch countdown {
+                case 1:
```
Comment (Erin-Mounts) [inline Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift:123]: keep in mind we'd need to update singular/plural selection code like this if we ever need to support a Slavic language, as they can have multiple different plural forms depending on the last digit of the number...

File: Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift [hunk 4]
```diff
@@ -134,16 +171,95 @@ open class RSDActiveStepViewController: RSDFullscreenImageStepViewController {
...
-            let stepDuration = self.activeStep?.duration,
-            let duration = clock?.runningDuration() {
+        if UIApplication.shared.applicationState != .active {
             self.countdownDial?.progress = CGFloat(duration / stepDuration)
         }
+        
+        // TODO: syoung 07/02/2019 Update implementation to include *restart* of the step should the step
+        // be reset and need the recorders to restart.
+
+        // If the timer duration is more than the step duration, then
```
Comment (Erin-Mounts) [inline Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift]: then what

**Review by Erin-Mounts — APPROVED** [review]
a few nits, otherwise looks good

File: Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift (hunk 3 above)
Comment (syoung-smallwisdom) [reply Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift:123]: Yes. *sigh* I cannot figure out the magic for getting `.stringsdict` files to work for other than the "Localizable.stringsdict" in the main bundle. Decided that it was a task for another day.

Really, this *should* use the built in function for time units, but *that* doesn't work for time. (works for other types of units).

**Pull request merged**